extensions/mssql/src/sparkFeature/dialog/sparkJobSubmission/sparkJobSubmissionModel.ts

```diff
@@ -0,0 +1,206 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as sqlops from 'sqlops';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+import * as fs from 'fs';
+import * as fspath from 'path';
+import * as os from 'os';
+
+import * as constants from '../../../constants';
+import { SqlClusterConnection } from '../../../objectExplorerNodeProvider/connection';
+import * as LocalizedConstants from '../../../localizedConstants';
+import * as utils from '../../../utils';
+import { SparkJobSubmissionService, SparkJobSubmissionInput, LivyLogResponse } from './sparkJobSubmissionService';
+import { AppContext } from '../../../appContext';
+import { IFileSource, File, joinHdfsPath } from '../../../objectExplorerNodeProvider/fileSources';
+
+
+// Stores important state and service methods used by the Spark Job Submission Dialog.
+export class SparkJobSubmissionModel {
+    public hadoopConnection: SqlClusterConnection;
+    private _dialogService: SparkJobSubmissionService;
+    private _guidForClusterFolder: string;
+    public get guidForClusterFolder(): string { return this._guidForClusterFolder; }
+
+    // Whether the file is from local or HDFS
+    public isMainSourceFromLocal: boolean;
+
+    // indicate the final path to be submitted within HDFS
+    public hdfsSubmitFilePath: string;
+
+    // local file uploading related path: source; destinationFolder
+    public localFileSourcePath: string;
+    public hdfsFolderDestinationPath: string;
+
+    constructor(
+        private readonly _sqlClusterConnection: sqlops.connection.Connection,
```

Comment:
As mentioned before, how about the input to the overall set of dialogs is the `SqlClusterConnection` object so we avoid all this confusion.

Reply:
fixed.